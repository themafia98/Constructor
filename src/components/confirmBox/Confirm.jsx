import React from 'react';
import PropTypes from 'prop-types';

import './Confirm.scss';

const Confirm = props => {

    return (
        <div className = 'wrapperConfirm'>
            <div className = 'confirmSave'>
                <span>Detected new changes</span>
                <input
                    className = 'saveCofirm'
                    onClick = {props.cbSaveChanges}
                    type = 'button'
                    value = 'Save changes'
                />
                <input
                    className = 'cancelCofirm'
                    onClick = {props.cbCancelSave}
                    type = 'button'
                    value = 'Cancel'
                />
            </div>
        </div>
    )
}

Confirm.propTypes = {
    cbSaveChanges: PropTypes.func.isRequired, // callback
    cbCancelSave: PropTypes.func.isRequired // callback
}

export default Confirm;