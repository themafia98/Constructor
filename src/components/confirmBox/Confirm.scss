.wrapperConfirm {
    position: fixed;
    top: 0;
    width: 100%;
    height: 100%;
    overflow: hidden;
    background-color: rgba(0, 0, 0, 0.6);
    z-index: 999;
}

.confirmSave {
    position: fixed;
    transform: translate(-50%,-50%);
    left: 50%;
    top: 50%;
    width: 200px;
    height: 110px;
    z-index: 9999;
    border: 1px solid black;
    background-color: #FFFF;
    border-radius: 10px;
    box-shadow: 0 0 5px;
    display: flex;
    flex-direction: column;
    span {
        text-align: center;
        font-weight: bold;
        padding: 5px;
    }

    .saveCofirm {
        border: 2px solid rgb(33, 44, 35);
        background: rgb(85, 236, 37);
        padding: 5px;
        margin:  0 10px;
        font-weight: bold;
    }
    .cancelCofirm {
        border: 2px solid rgb(33, 44, 35);
        background: rgb(228, 41, 8);
        padding: 5px;
        margin: 10px;
        font-weight: bold;
    }
}