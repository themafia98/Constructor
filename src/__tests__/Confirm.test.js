import React from 'react';
import renderer from 'react-test-renderer';
import Confirm from '../components/confirmBox/Confirm';

const saveChanges = props => {
    console.log('button save work!');
}

const cancelChanges = props => {
    console.log('cancel save work!');
}

const ConfirmComponent = renderer.create(
    <Confirm cbSaveChanges = {saveChanges} cbCancelSave = {cancelChanges} title = 'icon' />
);

test('Confirm test ', function () {

    let snaphotConfirmComponent = ConfirmComponent.toJSON();
    expect(snaphotConfirmComponent).toMatchSnapshot();

    const saveCofirm = ConfirmComponent.root.find(item => item.props.className === 'saveCofirm');
    saveCofirm.props.onClick({stopPropagation: () => null});

    snaphotConfirmComponent = ConfirmComponent.toJSON();
    expect(snaphotConfirmComponent).toMatchSnapshot();

    const cancelCofirm = ConfirmComponent.root.find(item => item.props.className === 'cancelCofirm');
    cancelCofirm.props.onClick({stopPropagation: () => null});

    snaphotConfirmComponent = ConfirmComponent.toJSON();
    expect(snaphotConfirmComponent).toMatchSnapshot();



});