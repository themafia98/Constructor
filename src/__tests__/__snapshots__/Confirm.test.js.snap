// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm test  1`] = `
<div
  className="wrapperConfirm"
>
  <div
    className="confirmSave"
  >
    <span>
      Detected new changes
    </span>
    <input
      className="saveCofirm"
      onClick={[Function]}
      type="button"
      value="Save changes"
    />
    <input
      className="cancelCofirm"
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
</div>
`;

exports[`Confirm test  2`] = `
<div
  className="wrapperConfirm"
>
  <div
    className="confirmSave"
  >
    <span>
      Detected new changes
    </span>
    <input
      className="saveCofirm"
      onClick={[Function]}
      type="button"
      value="Save changes"
    />
    <input
      className="cancelCofirm"
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
</div>
`;

exports[`Confirm test  3`] = `
<div
  className="wrapperConfirm"
>
  <div
    className="confirmSave"
  >
    <span>
      Detected new changes
    </span>
    <input
      className="saveCofirm"
      onClick={[Function]}
      type="button"
      value="Save changes"
    />
    <input
      className="cancelCofirm"
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
</div>
`;
